server/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -626,6 +628,11 @@ public void markResponseRestricted(String restriction) {
         consumedParams.add(RESPONSE_RESTRICTED);
     }
 
+    @Override
+    public SpanId getSpanId() {
+        return SpanId.forRestRequest(this);
```

Comment:
SpanId itself will need to be moved into the telemetry lib, so having static methods to construct a SpanId doesn't seem like it will work long term. What do you think about just making the SpanId ctor public and letting the implementation here choose how to construct it?

Reply:
++ on constructing SpanID in RestRequest/Task
alternatively we could have getSpandId default implementation and have a getRawSpanID method on the interface.
```
public interface Traceable {
    default SpanId getSpanId() {
        new SpanId(getRawSpanID());
    }
}
String getRawSpanID();
```